vcpkg/examples/assetcaching-script.md

```diff
@@ -0,0 +1,70 @@
+---
+title: Asset Caching with an alternate storage source
+description: Learn about adding a new storage source
+ms.date: 4/18/2023
+---
+
+# Asset Caching with an alternate storage source
+
+[!INCLUDE [experimental](../../includes/experimental.md)]
+
+Your scenario might need a different storage source such as NuGet feed, Universal Packages, or something else.
+
+In this example, we will set up a NuGet feed as a source, meaning we will use `nuget.exe` to pull and push assets.
+
+First, we will set up the environment before any call to `vcpkg install`:
+```powershell
+$env:X_VCPKG_ASSET_SOURCES="clear;x-script,C:/path/to/sources.bat {url} {sha512} {dst};x-block-origin"
+$env:NUGET=C:/path/to/nuget.exe
+$env:VCPKG_KEEP_ENV_VARS=NUGET
+```
+`X_VCPKG_ASSET_SOURCES` passes in a script `sources.bat` which will handle `nuget` push and pull commands. The `vcpkg` environment needs to use a copy of `nuget.exe` which can be supplied by the environment with [VCPKG_KEEP_ENV_VARS](config-environment.md#vcpkg_keep_env_vars).
+
+The script `sources.bat` checks if the asset is stored in the feed, and if so, installs it to the destination path. If the asset is not stored on the feed, the script fetches the url via curl, packs the asset in a `nuget` package and pushes it to the feed.
+
+
+`sources.bat`
```

Comment:
As a point of style, we should not introduce files by simply having their name in backticks and then the content -- instead, we should use prose to instruct the user to create a file name so-and-so.

(example from the dotnet docs: https://learn.microsoft.com/en-us/aspnet/core/tutorials/min-web-api?view=aspnetcore-7.0&tabs=visual-studio#the-model-and-database-context-classes)

```suggestion
Create a `sources.bat` file in your project with the following content:
```

This should probably be reordered above the environment setting steps above